core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java

```diff
@@ -92,8 +92,9 @@ public synchronized void heartbeat() throws InterruptedException {
         mContext.getClientContext().getPathConfHash());
     if (isClusterConfUpdated || isPathConfUpdated) {
       try {
-        mContext.reinit(isClusterConfUpdated, isPathConfUpdated);
-        mException = null;
+        if (mContext.reinit(isClusterConfUpdated, isPathConfUpdated)) {
+          mException = null;
+        }
```

Comment:
Some warning would be useful if not synced. Basically the same message as in the deleted TimeoutException section.